modules/twinkleunlink.js

```diff
@@ -14,15 +14,11 @@
  */
 
 Twinkle.unlink = function twinkleunlink() {
-	if( mw.config.get('wgNamespaceNumber') < 0 || mw.config.get('wgPageName') === 'Wikipedia:Sandbox' ) {
+	if( mw.config.get('wgNamespaceNumber') < 0 || mw.config.get('wgPageName') === 'Wikipedia:Sandbox' ||
+		!Morebits.userIsInGroup('extendedconfirmed')) {
 		return;
 	}
-	// Restrict to extended confirmed users. This is done here instead of above to avoid an unnecessary API call.
-	mw.user.getRights().then(function (rights) {
-		if( rights.indexOf('extendedconfirmed') !== -1 ) {
-			Twinkle.addPortletLink( Twinkle.unlink.callback, "Unlink", "tw-unlink", "Unlink backlinks" );
-		}
-	});
```

Comment:
Rights need to be checked, not groups. `sysop` has the `extendedconfirmed` right, but `sysop`s are not in the `extendedconfirmed` group.

Reply:
Ah, still we can manage without an API call.

Reply:
Just to note that this will disallow `bot`s, unlike the previous check.  I have a hard time imagining someone using a bot to unlink, much less doing so with Twinkle's unlink module, but that's the other group with the `extendedconfirmed` right.  I don't think it's necessary.